dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -440,6 +441,109 @@ def _lu_factor(a, res_type):
         return (a_h, ipiv_h, dev_info_array)
 
 
+def dpnp_cholesky_batch(a, res_type):
+    """
+    dpnp_cholesky_batch(a, res_type)
+
+    Return the batched Cholesky decomposition of `a` array.
+
+    """
+
+    a_sycl_queue = a.sycl_queue
+    a_usm_type = a.usm_type
+
+    n = a.shape[-2]
+
+    orig_shape = a.shape
+    # get 3d input arrays by reshape
+    a = a.reshape(-1, n, n)
+    batch_size = a.shape[0]
+    a_usm_arr = dpnp.get_usm_ndarray(a)
+
+    # `a` must be copied because potrf_batch destroys the input matrix
+    a_h = dpnp.empty_like(a, order="C", dtype=res_type, usm_type=a_usm_type)
+
+    # use DPCTL tensor function to fill the сopy of the input array
+    # from the input array
+    a_ht_copy_ev, a_copy_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+        src=a_usm_arr, dst=a_h.get_array(), sycl_queue=a_sycl_queue
+    )
+
+    a_stride = a_h.strides[0]
+
+    # Call the LAPACK extension function _potrf_batch
+    # to computes the Cholesky decomposition of a batch of
+    # symmetric positive-definite matrices
+    ht_lapack_ev, _ = li._potrf_batch(
+        a_sycl_queue,
+        a_h.get_array(),
+        n,
+        a_stride,
+        batch_size,
+        [a_copy_ev],
+    )
+
+    ht_lapack_ev.wait()
+    a_ht_copy_ev.wait()
+
+    a_h = dpnp.tril(a_h.reshape(orig_shape))
+
+    return a_h
+
+
+def dpnp_cholesky(a):
+    """
+    dpnp_cholesky(a)
+
+    Return the Cholesky decomposition of `a` array.
+
+    """
+
+    a_sycl_queue = a.sycl_queue
+    a_usm_type = a.usm_type
+
+    res_type = _common_type(a)
+
+    a_shape = a.shape
+
+    if a.size == 0:
+        return dpnp.empty(
+            a_shape,
+            dtype=res_type,
+            usm_type=a_usm_type,
+            sycl_queue=a_sycl_queue,
+        )
+
+    if a.ndim > 2:
+        return dpnp_cholesky_batch(a, res_type)
+
+    a_usm_arr = dpnp.get_usm_ndarray(a)
+
+    # `a` must be copied because potrf destroys the input matrix
+    a_h = dpnp.empty_like(a, order="C", dtype=res_type, usm_type=a_usm_type)
+
+    # use DPCTL tensor function to fill the сopy of the input array
+    # from the input array
+    a_ht_copy_ev, a_copy_ev = ti._copy_usm_ndarray_into_usm_ndarray(
+        src=a_usm_arr, dst=a_h.get_array(), sycl_queue=a_sycl_queue
+    )
+
+    # Call the LAPACK extension function _potrf
+    # to computes the Cholesky decomposition
+    ht_lapack_ev, _ = li._potrf(
+        a_sycl_queue,
+        a_h.get_array(),
+        [a_copy_ev],
+    )
+
+    ht_lapack_ev.wait()
+    a_ht_copy_ev.wait()
+
+    a_h = dpnp.tril(a_h)
```

Comment:
why do we need to create a copy here? can we control the required decomposition by `upper_lower` flag and to avoid the extra copying?

Reply:
Because we need to get `lower` or `upper` triangular matrix with zeros `above` or `below` the main diagonal.
Since `potrf` and `potrf_batch` writes the result to the input array, we need to get rid of the input array data that has not changed.